### DISCUSSION TITLE
Header authentication

### DISCUSSION THREAD

**Discussion by skaravad** [tchiotludo/akhq#718] [Q&A]
Hello All,
I see that AKHQ supports JWT authentication but the documentation is very limited, wondering if anyone tried it , on a side note, are there any plans to support Header based Authentication and Authorization , for instance we can set User and User group(s) via reverse proxy (see vouch proxy https://github.com/vouch/vouch-proxy) and headers are forwarded to the backend.

Using the above method I can forward the entire JWT token to AKHQ , but wondering how to proceed after that, any help is much appreciated.

**Comment by tchiotludo** [comment]
Hello, 
You want to use a reverse proxy for auth and just forward an already forge JWT to akhq ? 
Seems to be tricky since the JWT token is internal to AKHQ and contains some role inside that I can change in any update. 

In a meantime, I have an idea to allow passing a custom header from the reverse prooxy that will contains the username, AKHQ will read this users and handle mapping with akhq configuration.

**Reply by skaravad** [reply]
@tchiotludo  thank you, If the header model works , then it would be ideal, we can set headers (map) with user, group etc which can make the authentication/authorization much simpler.

**Reply by tchiotludo** [reply]
I've added this issue: https://github.com/tchiotludo/akhq/issues/724.
You can add some feedback for your use case
